File: superhero/src/main/java/com/frontguys/superhero/models/Request.java
package com.frontguys.superhero.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;

public class Request {
    private int id;
    private Integer customerId;
    private ClientDetails customerDetails;
    private Integer contractorId;
    private ClientDetails contractorDetails;
    private String type;
    private String address;
    private Date expirationDate;
    private Date publishDate;
    private boolean isFinishedByCustomer;
    private boolean isFinishedByContractor;
    private boolean isApproved;
    private int responseCount;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Integer getCustomerId() {
        return customerId;
    }

    public void setCustomerId(Integer customerId) {
        this.customerId = customerId;
    }

    public ClientDetails getCustomerDetails() {
        return customerDetails;
    }

    public void setCustomerDetails(ClientDetails customerDetails) {
        this.customerDetails = customerDetails;
    }

    public Integer getContractorId() {
        return contractorId;
    }

    public void setContractorId(Integer contractorId) {
        this.contractorId = contractorId;
    }

    public ClientDetails getContractorDetails() {
        return contractorDetails;
    }

    public void setContractorDetails(ClientDetails contractorDetails) {
        this.contractorDetails = contractorDetails;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Date getExpirationDate() {
        return expirationDate;
    }

    public void setExpirationDate(Date expirationDate) {
        this.expirationDate = expirationDate;
    }

    public Date getPublishDate() {
        return publishDate;
    }

    public void setPublishDate(Date publishDate) {
        this.publishDate = publishDate;
    }

    @JsonProperty(value = "isFinishedByCustomer")
    public boolean isFinishedByCustomer() {
        return isFinishedByCustomer;
    }

    public void setFinishedByCustomer(boolean finishedByCustomer) {
        isFinishedByCustomer = finishedByCustomer;
    }

    @JsonProperty(value = "isFinishedByContractor")
    public boolean isFinishedByContractor() {
        return isFinishedByContractor;
    }

    public void setFinishedByContractor(boolean finishedByContractor) {
        isFinishedByContractor = finishedByContractor;
    }

    @JsonProperty(value = "isApproved")
    public boolean isApproved() {
        return isApproved;
    }

    public void setApproved(boolean approved) {
        isApproved = approved;
    }

    public int getResponseCount() {
        return responseCount;
    }

    public void setResponseCount(int responseCount) {
        this.responseCount = responseCount;
    }
}

File: superhero_flutter/lib/models/request.dart
import 'package:superhero_flutter/constants/request_types.dart';
import 'package:superhero_flutter/models/client_details.dart';

class Request {
  int id;
  int customerId;
  ClientDetails customerDetails;
  int contractorId;
  ClientDetails contractorDetails;
  String type;
  String typeValue;
  String address;
  String expirationDate;
  String publishDate;
  bool isFinishedByCustomer;
  bool isFinishedByContractor;
  bool isApproved;
  int responseCount;

  Request.fromJson(map) {
    id = map['id'];
    customerId = map['customerId'];
    customerDetails = ClientDetails.fromJson(map['customerDetails']);
    contractorId = map['contractorId'];
    contractorDetails = contractorId == null
        ? null
        : ClientDetails.fromJson(map['customerDetails']);
    type = map['type'];
    typeValue = RequestTypes.getTypeValue(type);
    address = map['address'];
    expirationDate = map['expirationDate'];
    publishDate = map['publishDate'];
    isFinishedByCustomer = map['isFinishedByCustomer'];
    isFinishedByContractor = map['isFinishedByContractor'];
    isApproved = map['isApproved'];
    responseCount = map['responseCount'];
  }
}
